GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/NotifyUsersAndRetireInactiveAccountsJob.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.Extensions;
+using GenderPayGap.WebUI.Services;
+
+namespace GenderPayGap.WebUI.BackgroundJobs.ScheduledJobs
+{
+    public class NotifyUsersAndRetireInactiveAccountsJob
+    {
+        private readonly IDataRepository dataRepository;
+        private readonly EmailSendingService emailSendingService;
+
+        public NotifyUsersAndRetireInactiveAccountsJob(IDataRepository dataRepository, EmailSendingService emailSendingService)
+        {
+            this.dataRepository = dataRepository;
+            this.emailSendingService = emailSendingService;
+        }
+
+        public void NotifyUsersAndRetireInactiveAccounts()
+        {
+            var runId = JobHelpers.CreateRunId();
+            var startTime = VirtualDateTime.Now;
+            JobHelpers.LogFunctionStart(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+
+            try
+            {
+                SendReminderEmails();
+
+                RetireUsers();
+
+                JobHelpers.LogFunctionEnd(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+            }
+            catch (Exception ex)
+            {
+                JobHelpers.LogFunctionError(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime, ex);
+
+                //Rethrow the error
+                throw;
+            }
+        }
+
+        private void SendReminderEmails()
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+            DateTime threeYearsAgoMinusThirtyDays = threeYearsAgo.AddDays(30);
+            DateTime threeYearsAgoMinusSevenDays = threeYearsAgo.AddDays(7);
+
+            List<User> usersToSendReminders = dataRepository.GetAll<User>()
+                .Where(u => u.LoginDate == threeYearsAgoMinusThirtyDays || u.LoginDate == threeYearsAgoMinusSevenDays)

```

Comment:
This isn't going to work.
`LoginDate` and `threeYearsAgoMinusX` are both Date**Time**s (they have both a date and time component), so this will only find users who's last login was (3 years - X days) ago to the millisecond.

We'll have to do something like (pseudocode):
```
Where Login >= X && Login < (X.AddDays(1))
```